std/encoding.d

```diff
@@ -2233,7 +2101,7 @@ body
     }
     else static if(is(Src==AsciiChar))
     {
-        transcode!(char,Dst)(cast(string)s,r);
+        transcodeFromAscii(s, r);
```

Comment:
It looks like all possible instantiations of `transcode` are memory safe. In that case `@trusted` should be on `transcode` in its entirety.